core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java

```diff
@@ -295,16 +296,28 @@ public Optional<FateTxStore<T>> createAndReserve(FateKey fateKey) {
     // This will create the FateTxStore before creation but this object
     // is not exposed until after creation is finished so there should not
     // be any errors.
+    final Optional<FateTxStore<T>> reservedTxStore;
     synchronized (this) {
-      txStore = tryReserve(fateId);
+      reservedTxStore = tryReserve(fateId);
     }
 
-    if (txStore.isPresent()) {
+    // If present we were able to reserve so try and create
+    if (reservedTxStore.isPresent()) {
       try {
-        Preconditions.checkState(create(fateKey) != null,
-            "Unexpected null FateId when creating and reserving fateKey %s", fateKey);
+        if (create(fateKey).isPresent()) {
+          txStore = reservedTxStore;
```

Comment:
Could modify the code to keep the return value of create and validate it.

```suggestion
           Precondition.checkState(fateId.equals(fateIdFromCreate));
          txStore = reservedTxStore;
```